app/models/deploy_group.rb

```diff
@@ -36,6 +36,10 @@ def natural_order
     name.split(/(\d+)/).each_with_index.map { |x, i| i.odd? ? x.to_i : x }
   end
 
+  def cloned_stages
```

Comment:
better to make this a scope on stage `scope :cloned, -> { where.not(template_stage_id: nil) }` then use group.stages.cloned